File: 3-SAT/FormulaCreator/ITriplet.cs
namespace FormulaCreator
{
    public interface ITriplet
    {
        /// <summary>
        /// One-based index of first variable in the triplet.
        /// </summary>
        int AName { get; }
        int BName { get; }
        int CName { get; }
        /// <summary>
        /// Value <code>true</code> means the variable is with logical NOT.
        /// </summary>
        bool NotA { get; }
        bool NotB { get; }
        bool NotC { get; }

        string CanonicalName { get; }
        bool HasSameVariablesAs(ITriplet triplet);
        ITriplet TransformAs(ITriplet triplet);
        bool HasVariable(int varName);
    }
}

File: 3-SAT/3-sat/3-sat-core/src/main/java/com/anjlab/sat3/ITriplet.java
package com.anjlab.sat3;

public interface ITriplet extends ITripletPermutation, ITripletValue
{
    /// <summary>
    /// Value <code>true</code> means the variable is with logical NOT.
    /// </summary>
    boolean isNotA();
    boolean isNotB();
    boolean isNotC();

    void transposeTo(ITripletPermutation targetPermutation);
}